### PR TITLE
[13.0] [MIG] project_task_digitized_signature

### PR DESCRIPTION
**Pull request by bodedra**
Here is TODO list for replicate same behavior in Sale Order:

-   [ ] Add signature template in Task portal view
-   [ ] Send "Print Task" report in a Task chatter area.

### CODE REVIEW

**Comment by bodedra** [comment]
@cubells @eLBati @sysadminmatmoz @pedrobaeza I have remove dependency of web_widget_digitized_signature module. It works great. I can sign in Task and print Task report.

Now in this PR, we are facing issue with tracking message. So may we remove this tracking message feature because we remove dependency of web_widget_**** module.

Any suggestions are welcome.

**Comment by bodedra** [comment]
Tested in runbot.

Screenshot from 2021-03-24 10-59-56

**Comment by pedrobaeza** [comment]
The document to mimic was sales order. Is this document now posting the signature in the thread? If so, just see how it's done. If not, it's OK to remove such thing.

**Comment by bodedra** [comment]
> The document to mimic was sales order. Is this document now posting the signature in the thread? If so, just see how it's done. If not, it's OK to remove such thing.

Nope. This document doesn't post the signature in the thread. I will remove that code from test case. So this PR become green.

JFI, I haven't reviewed posting signature in the sale order.

**Comment by bodedra** [comment]
@pedrobaeza From portal, when customer signs in SO, it posts PDF in a chatter with content "Order signed by ***Customer Name***" plus PDF report. But no such feature in sale order form level. In SO form, just allow attached images in "Customer Signature" tab.

**Comment by pedrobaeza** [comment]
OK, then we can replicate the same behavior here.

**Comment by bodedra** [comment]
Here is TODO list for replicate same behavior:

- Add signature template in Task portal view
- Send "Print Task" report in a Task chatter area.

**Comment by bodedra** [comment]
I have short deadline and will survive for now with as of now functionality. I will continue or contribute later on.

If anyone interested to take on, please feel free to do so.

Thank you @pedrobaeza for your valuable input.

**Comment by pedrobaeza** [comment]
OK, you can add to the ROADMAP that TODO list. Meanwhile, green Travis is required at least.

**Comment by bodedra** [comment]
> OK, you can add to the ROADMAP that TODO list. Meanwhile, green Travis is required at least.

It's :green_circle: now.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Several things to do yet:

- Pre-commit stuff is not split from the initial migration commit, so everything is mixed.
- Some commit squashing is needed.
- Update README according new implementation.

**Comment by dalonsod** [comment]
Hello @pedrobaeza this PR seems to be idle. I'm interested to make v14 migration of this addon, since I should need it in that version, not v13. Is a good approach doing, starting from this PR, the pending tasks you noticed, preserving authority, and then v14 tasks together in a v14 migration PR? Or directly make v12 to v14 migration that could be backported to v13 later, if needed? Or anything else... Thanks!

**Comment by pedrobaeza** [comment]
You can go directly from v12 to v14

**Comment by bodedra** [comment]
@pedrobaeza I not only migrated the module but also made it compatible with v13.0. Because there are many changes in base v13.0

@dalonsod  It is best to attend a few @pedrobaeza 's [comment](https://github.com/OCA/project/pull/802#pullrequestreview-621539430) and sign off v13.0 migration and migrate it in v14.0.

**Comment by pedrobaeza** [comment]
OK then, but that steps are needed then.

**Comment by bodedra** [comment]
Yup. I agree. That's why I haven't closed this PR.

**Comment by dalonsod** [comment]
Migration proposal for v14 available in #943 . @bodedra your improvements are covered and contributors mention included

**Comment by bodedra** [comment]
@dalonsod Wrong migration. I mentioned in https://github.com/OCA/project/pull/802#issuecomment-1176533994 that we will address @pedrobaeza 's comment and complete v13.0 migration first. And then v14.0.

If you don't want to address the above comment, then do the following away v14.0 module migration.
- Check out 14.0 branch
- cherry-pick v13.0 migration commits
- Perform 14.0 module migration

In that way, we will preserve committed history.

**Comment by dalonsod** [comment]
Hello @bodedra thanks for your comments. We simply followed original suggestion to direct migration from v12 to v14, in order to unblock this, we read your comments later, sorry.

But I'm a little bit confused, when you say "_that we will address @pedrobaeza 's comment and complete v13.0 migration first_", you'll saying you'll resume this? If it's right, we'll wait for it and when it's done, we'll make v14 migration.

**Comment by bodedra** [comment]
@dalonsod No problem. I am sorry I do not have any availability to continue development work.

You can take over this PR with cherry-pick and I explain both options in above comment.

**Comment by dalonsod** [comment]
OK thanks @bodedra we'll continue the migration with your comments

**Comment by dalonsod** [comment]
Hello @bodedra comments attended, you can check #943 v14 migration update, could you review?

imagen

* Made initial separated v13.0 pre-commit.
* Cherry-picked your v13 migration commits from here, squashed and preserved authorship. Also updated `INSTALL` file that still referred `web_widget_digitized_signature`, removing it.
* And at last, pre-commit and migration commits for v14.

**Comment by dalonsod** [comment]
Hello @bodedra could you approve this PR and #943 ? I've made the required changes for both migrations, as suggested.

Thanks in advance!

**Comment by bodedra** [comment]
Thank you @dalonsod for your contribution. I cannot approve my own PR. I reviewed code again and looks good. 

You can review this and approve it.

**Comment by bodedra** [comment]
@OCA/project-service-maintainers Would you please review it?

File: project_task_digitized_signature/tests/test_project_task_digitized_signature.py
Comment (pedrobaeza) [inline project_task_digitized_signature/tests/test_project_task_digitized_signature.py]: This test is now useless, as it's not the testing anything related to the module.

**Review by pedrobaeza — APPROVED** [review]
The scope is very limited, but as added in the ROADMAP, let's not block this anymore

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Wait! The dependency over `project_task_report` is overkilled. This base module mustn't depend on the other.

**Review by pedrobaeza — APPROVED** [review]
As talked in the 14.0 PR, the dependency is from the beginning, so let's not block this again:

/ocabot merge nobump

**Pull request merged**
